### ISSUE TITLE
GHPoint violates equals/hashCode contract

### ISSUE THREAD

**Issue by easbar** [graphhopper/graphhopper#2445]
```java
        GHPoint a = new GHPoint(51.9855002, 19.2540004);
        GHPoint b = new GHPoint(51.9855, 19.254);
        System.out.println(a.hashCode());
        System.out.println(b.hashCode());
        System.out.println(a.equals(b));
```

prints: 

```
-703473411
1833648456
true
```        

I wonder if we should just remove equals/hashCode from GHPoint. Which points are supposed to be treated equal or different depends on the specific application anyway.

**Comment by michaz** [comment]
I agree with everyone and everything. Some thoughts (maybe additional, maybe redundant):

- We can/should not only emulate, but *migrate* to JTS `Coordinate` wherever we come across it and it's convenient. No rush, but this was already agreed at some point.
- *I do this, too*, but in theory it's already considered Bad to use `doubles` as hashmap-keys. (Exactly because, in practice, equality _in the domain_ is mostly "up to epsilon") So probably double-vectors even more so. :)
